includes/payment-methods/class-wc-stripe-upe-payment-gateway.php

```diff
@@ -2132,4 +2142,66 @@ private function get_return_url_for_redirect( $order, $save_payment_method ) {
 			)
 		);
 	}
+
+	/**
+	 * Retrieves the (possible) existing payment intent for an order and payment method types.
+	 *
+	 * @param WC_Order $order The order.
+	 * @param array $payment_method_types The payment method types.
+	 * @return object|null
+	 * @throws WC_Stripe_Exception
+	 */
+	private function get_existing_compatible_payment_intent( $order, $payment_method_types ) {
+		// Reload the order to make sure we have the latest data.
+		$order  = wc_get_order( $order->get_id() );
```

Comment:
I'm curious about this approach. What's the difference from using the `$order` variable that's passed to this method?